.github/workflows/cxx.yml

```diff
@@ -0,0 +1,247 @@
+          - { os: windows-2019, target: g++-6 }
+          - { os: windows-2019, target: g++-5 }
+
+          # TODO: Implement support for clang and MSVC on Windows
+          # Currently those are still being run by the auto-tester
+          - os: windows-2019
+
+        include:
+          # Clang boilerplate
+          - { target: clang-9.0.0, compiler: clang, cxx-version: 9.0.0 }
+          - { target: clang-8.0.0, compiler: clang, cxx-version: 8.0.0 }
+          - { target: clang-7.0.0, compiler: clang, cxx-version: 7.0.0 }
+          - { target: clang-6.0.0, compiler: clang, cxx-version: 6.0.0 }
+          - { target: clang-5.0.2, compiler: clang, cxx-version: 5.0.2 }
+          - { target: clang-4.0.0, compiler: clang, cxx-version: 4.0.0 }
+          - { target: clang-3.9.0, compiler: clang, cxx-version: 3.9.0 }
+          # g++ boilerplace
+          - { target: g++-9, compiler: g++, cxx-version: 9.3.0 }
+          - { target: g++-8, compiler: g++, cxx-version: 8.4.0 }
+          - { target: g++-7, compiler: g++, cxx-version: 7.5.0 }
+          - { target: g++-6, compiler: g++, cxx-version: 6.5.0 }
+          - { target: g++-5, compiler: g++, cxx-version: 5.5.0 }
+          # Platform boilerplate
+          - { os: ubuntu-16.04, arch: x86_64-linux-gnu-ubuntu-16.04 }
+          - { os: macOS-10.15,  arch: x86_64-apple-darwin }
+          # Clang 9.0.0 have a different arch for OSX
+          - { os: macOS-10.15, target: clang-9.0.0, arch: x86_64-darwin-apple }
+
+    runs-on: ${{ matrix.os }}
+    steps:
+
+    ########################################
+    #    Setting up the host D compiler    #
+    ########################################
+    - name: Prepare compiler
+      uses: mihails-strasuns/setup-dlang@v0.5.0
+      with:
+          compiler: dmd-2.091.0
+          gh_token: ${{ secrets.GITHUB_TOKEN }}
+
+    #########################################
+    # Checking out up DMD, druntime, Phobos #
+    #########################################
+    - name: Checkout DMD
+      uses: actions/checkout@v2
+      with:
+        path: dmd
+        repository: dlang/dmd
+        ref: ${{ github.base_ref }}
+        persist-credentials: false
+    - name: Checkout druntime
+      uses: actions/checkout@v2
+      with:
+        path: druntime
+        persist-credentials: false
+    - name: Checkout Phobos
+      uses: actions/checkout@v2
+      with:
+        path: phobos
+        repository: dlang/phobos
+        ref: ${{ github.base_ref }}
+        persist-credentials: false
+
+
+    ########################################
+    #   Setting up the host C++ compiler   #
+    ########################################
+    - name: '[Posix] Setting up clang ${{ matrix.cxx-version }}'
+      if: matrix.compiler == 'clang' && runner.os != 'Windows'
+      run: |
+        wget --quiet --directory-prefix=${{ github.workspace }} https://releases.llvm.org/${{ matrix.cxx-version }}/clang+llvm-${{ matrix.cxx-version }}-${{ matrix.arch }}.tar.xz
```

Comment:
(I'm on the phone, so I can't check, but) it seems like this very similar or the same as script inside the dmd GitHub Actions config. What do you think about extracting it to dmd/ci/setup_cxx.bash?

AFAIR we're already doing something similar for CircleCI in phobos.

Reply:
We can't extract it because github pre-process those scripts, e.g. the `${{ github.workspace }}` is replaced to create a script.
Of course I could use the environment variables, but it looks like a step back in readability.
I think a better course of action would be to extract this into its own separate action, which I'll do once the Windows support is ready.

Reply:
Yes, extracting it as an action (or a Docker image) that can be reused actually repos is u ultimately the best way to go. I didn't propose this as I thought it would require much more work from your side.